Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/InputActionsEditorView.cs

```diff
@@ -159,6 +169,49 @@ public class ViewState
             public IEnumerable<InputControlScheme> controlSchemes;
             public int selectedControlSchemeIndex;
         }
+
+        void OnExecuteCommand(ExecuteCommandEvent evt)
```

Comment:
Feels like double work - do we handle events twice here? (additional to handling them in the InputActionsTreeView / InputActionMapsView)

Reply:
If you mean, could both be called with one action - I don't believe this is possible since we StopPropagation.

If you mean there's code overlap - I think I've now reduced the code in this section to the point where reducing it further would just mean adding an extra string comparison and function call to jump to the maps/actions views' OnExecuteCommand calls, but let me know what you think 🙂

On a somewhat related note to that - do we need the HasPastableClipboardData calls in OnExecuteCommand in ActionsTreeView/ActionMapsView? Won't that be checked anyway in one of the functions that PasteItems calls? It feels like that's logically where it should happen, if it doesn't already.

Reply:
Is it? Even with using the events on TrickleDown? If the second event is used OnBubbleUp this means we could catch the event beforehand. 
I thought about a delegation concept when I said "do we handle events twice", a event handler somewhere that delegates to the list/treeView. It was more a rough idea, if the trickledown/bubbleup does not work out. 
About your second comment: I think there is a safety check, and if we could indeed skip the one in the list/treeviews.

Reply:
Ah - were you suggesting removing the Paste event handlers for the Action Maps/Actions, and having a single handler for the entire window? And then events from anywhere lower in the hierarchy can bubble up to the root?

Reply:
yes, that was what I was thinking of as one solution. The other way would be to catch the events at actionMap/actionTreeView on trickleDown, and stop the event propagation and then only if nothing was registered there we catch the rest of the events on bubble up on the root element to handle them from there. 
For your reference: https://docs.unity3d.com/2021.2/Documentation/Manual/UIE-Events-Handling.html
This describes the event phases pretty well.

Reply:
I think like that the logic to paste actions into action maps (and add them as last element) is broken because it is impossible to determine from outside of the actionMapView in wich list the paste was performed. Added @Pauliusd01